crates/circuit/src/parameter_expression.rs

```diff
@@ -10,85 +10,1250 @@
+impl Add for &ParameterExpression {
+    type Output = ParameterExpression;
+    #[inline]
+    fn add(self, other: &ParameterExpression) -> Self::Output {
+        self._add(other)
+    }
+}
+
+macro_rules! add_impl_expr {
+    ($($t:ty)*) => ($(
+        impl Add<$t> for &ParameterExpression {
+            type Output = ParameterExpression;
+
+            #[inline]
+            #[track_caller]
+            fn add(self, other: $t) -> Self::Output {
+                self._add(&other.into())
+            }
+        }
+    )*)
+}
+
+add_impl_expr! {f64 i32 u32}
+
+// =============================
+// Sub
+// =============================
+impl Sub for &ParameterExpression {
+    type Output = ParameterExpression;
+    #[inline]
+    fn sub(self, other: &ParameterExpression) -> Self::Output {
+        self._sub(other)
+    }
+}
+
+macro_rules! sub_impl_expr {
+    ($($t:ty)*) => ($(
+        impl Sub<$t> for &ParameterExpression {
+            type Output = ParameterExpression;
+
+            #[inline]
+            #[track_caller]
+            fn sub(self, other: $t) -> Self::Output {
+                self._sub(&other.into())
+            }
+        }
+    )*)
+}
+
+sub_impl_expr! {f64 i32 u32}
+
+// =============================
+// Mul
+// =============================
+impl Mul for &ParameterExpression {
+    type Output = ParameterExpression;
+    #[inline]
+    fn mul(self, other: &ParameterExpression) -> Self::Output {
+        self._mul(other)
+    }
+}
+
+macro_rules! mul_impl_expr {
+    ($($t:ty)*) => ($(
+        impl Mul<$t> for &ParameterExpression {
+            type Output = ParameterExpression;
+
+            #[inline]
+            #[track_caller]
+            fn mul(self, other: $t) -> Self::Output {
+                self._mul(&other.into())
+            }
+        }
+    )*)
+}
+
+mul_impl_expr! {f64 i32 u32}
+
+// =============================
+// Div
+// =============================
+impl Div for &ParameterExpression {
+    type Output = ParameterExpression;
+    #[inline]
+    fn div(self, other: &ParameterExpression) -> Self::Output {
+        self._div(other)
+    }
+}
+
+macro_rules! div_impl_expr {
+    ($($t:ty)*) => ($(
+        impl Div<$t> for &ParameterExpression {
+            type Output = ParameterExpression;
+
+            #[inline]
+            #[track_caller]
+            fn div(self, other: $t) -> Self::Output {
+                self._div(&other.into())
+            }
+        }
+    )*)
+}
+div_impl_expr! {f64 i32 u32}
+
+// =================================================================
+//   Python interface to ParameterExpression/Parameter/ParameterVectorElement
+// =================================================================
+#[pymethods]
+impl ParameterExpression {
+    /// ParameterExpression::__init__
+    /// initialize ParameterExpression from the equation stored in string
+    #[new]
+    #[pyo3(signature = (symbol_map = None, expr = None, _qpy_replay = None))]
```

Comment:
```suggestion
    #[pyo3(signature = (symbol_map = None, expr = None, *, _qpy_replay = None))]
```

I think it's best to make sure the `_qpy_replay` is kwarg only.